app/src/main/java/io/homeassistant/companion/android/controls/HaControlsProviderService.kt

```diff
@@ -140,4 +208,26 @@ class HaControlsProviderService : ControlsProviderService() {
             consumer.accept(ControlAction.RESPONSE_UNKNOWN)
         }
     }
+
+    private fun getAreaForEntity(
+        entityId: String,
+        areaRegistry: List<AreaRegistryResponse>,
+        deviceRegistry: List<DeviceRegistryResponse>,
+        entityRegistry: List<EntityRegistryResponse>
+    ): AreaRegistryResponse? {
+        val rEntity = entityRegistry.firstOrNull { it.entityId == entityId }
+        if (rEntity != null) {
+            // By default, an entity should be considered to be in the same area as the associated device (if any)
+            // This can be overridden for an individual entity, so check the entity registry first
+            if (rEntity.areaId != null) {
+                return areaRegistry.firstOrNull { it.areaId == rEntity.entityId }
```

Comment:
I'll admit to not following exactly what is going on here, but should this really be comparing area and entity ids?  At first glance those don't seem to be comparable.

Reply:
That should indeed be comparing the area id, fixed!